server/src/main/scala/services/prediction/SplitsPredictor.scala

```diff
@@ -0,0 +1,206 @@
+package services.prediction
+
+import drt.shared._
+import drt.shared.SplitRatiosNs.SplitSources
+import org.apache.spark.ml.linalg.Vectors
+import org.apache.spark.ml.regression.{LinearRegression, LinearRegressionModel}
+import org.apache.spark.sql.{DataFrame, SparkSession}
+import org.apache.spark.sql.functions.{col, concat_ws, expr}
+import org.slf4j.{Logger, LoggerFactory}
+import passengersplits.core.SplitsCalculator
+import services.{SDate, SplitsProvider}
+import services.SplitsProvider.SplitProvider
+import services.graphstages.FeatureSpec
+
+sealed trait SplitsPredictorLike {
+  def predictForArrivals(arrivals: Seq[Arrival]): Seq[(Arrival, Option[ApiSplits])]
+}
+
+case class SplitsPredictor(sparkSession: SparkSession, portCode: String, flightCodes: Set[String], splitsView: DataFrame)
+  extends SplitsPredictorLike {
+  val log: Logger = LoggerFactory.getLogger(getClass)
+
+  val splitTypeToQueue = Map(
+    "EeaMachineReadable" -> Queues.EeaDesk,
+    "EeaNonMachineReadable" -> Queues.EeaDesk,
+    "VisaNational" -> Queues.NonEeaDesk,
+    "NonVisaNational" -> Queues.NonEeaDesk)
+
+  val paxTypes: List[String] = splitTypeToQueue.keys.toList
+
+  val featureSpecs = List(
+    FeatureSpec(List("flight", "day"), "fd"),
+    FeatureSpec(List("flight", "month"), "fm"),
+    FeatureSpec(List("flight", "origin"), "fo"),
+    FeatureSpec(List("day"), "d"),
+    FeatureSpec(List("month"), "m"),
+    FeatureSpec(List("origin"), "o")
+  )
+
+  splitsView.createOrReplaceTempView("splits")
+
+  val historicalSplitsProvider: SplitProvider = SplitsProvider.csvProvider
+
+  val splitsCalculator = SplitsCalculator(portCode, historicalSplitsProvider, Set())
```

Comment:
pass this in rather.

Reply:
turns out it wasn't used so have removed, along with the historic spits provider